src/contract/assembled_transaction.ts

```diff
@@ -836,28 +846,40 @@ export class AssembledTransaction<T> {
 
     // eslint-disable-next-line no-restricted-syntax
     for (const [i, entry] of authEntries.entries()) {
+      // workaround for https://github.com/stellar/js-stellar-sdk/issues/1070
+      const credentials = xdr.SorobanCredentials.fromXDR(entry.credentials().toXDR())
       if (
-        entry.credentials().switch() !==
+        credentials.switch() !==
         xdr.SorobanCredentialsType.sorobanCredentialsAddress()
       ) {
         // if the invoker/source account, then the entry doesn't need explicit
         // signature, since the tx envelope is already signed by the source
         // account, so only check for sorobanCredentialsAddress
         continue; // eslint-disable-line no-continue
       }
-      const pk = StrKey.encodeEd25519PublicKey(
-        entry.credentials().address().address().accountId().ed25519(),
-      );
+      const authEntryAddress = Address.fromScAddress(
+        credentials.address().address(),
+      ).toString();
 
       // this auth entry needs to be signed by a different account
       // (or maybe already was!)
-      if (pk !== publicKey) continue; // eslint-disable-line no-continue
+      if (authEntryAddress !== address) continue; // eslint-disable-line no-continue
+
+      const sign = signAuthEntry ??
+        function fakeSignAuthEntry(x) {
+          return Promise.resolve(x);
+        }
```

Comment:
```suggestion
      const sign = signAuthEntry ?? Promise.resolve;
```
Does this work?

Reply:
Need to type it so TS knows which overload of `Promise.resolve` to use, but it's still better than what I had. Thanks! 

```ts
      const sign: typeof signAuthEntry = signAuthEntry ?? Promise.resolve;
```